lib/ansible/module_utils/cloud.py

```diff
@@ -59,15 +59,15 @@ def status_code_from_exception(error):
         pass
 
     @staticmethod
-    def found(response_code):
+    def found(response_code, added_exceptions):
         """ Return True if the Response Code to retry on was found.
         Args:
             response_code (str): This is the Response Code that is being matched against.
         """
         pass
 
     @classmethod
-    def backoff(cls, tries=10, delay=3, backoff=1.1):
+    def backoff(cls, tries=10, delay=3, backoff=1.1, added_exceptions=[]):
```

Comment:
Declaring a method with a pointer-ish type (like list) as the default can result in some unexpected behavior.

```
>>> class Fooo(object): 
...     def bar(self, fizz=[]):                 
...         print(self, fizz)                   
...         fizz.append('a')                    
>>> f = Fooo()          
>>> f.bar()             
(<Fooo object at 0x7f5a61543290>, [])           
>>> f.bar()             
(<Fooo object at 0x7f5a61543290>, ['a'])        
>>> f.bar()             
(<Fooo object at 0x7f5a61543290>, ['a', 'a'])   
```

Instead, we can use `None` as the default and then check `if added_exceptions is not None:` to decide whether to extend the default exception list or not.